openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchIndexSink.java

```diff
@@ -33,29 +36,80 @@ public BulkResponse write(BulkRequest data, Map<String, Object> contextData)
       throws SearchIndexException {
     LOG.debug("[EsSearchIndexSink] Processing a Batch of Size: {}", data.numberOfActions());
     try {
-      BulkResponse response = searchRepository.getSearchClient().bulk(data, RequestOptions.DEFAULT);
-      int currentSuccess = getSuccessFromBulkResponseEs(response);
-      int currentFailed = response.getItems().length - currentSuccess;
+      BulkResponse response = null;
+      int offset = 0;
```

Comment:
what is offset trackign? is it the no.of entities we are going to index?

Reply:
purpose of offset tracking is to maintain the bufferData until it has not reached the payload limit inside the batch.

Reply:
@harshach number of records reaching sink are batched on the basis of their size , offset is used to track the records in the main list that sink receives

Reply:
batchSize -> Number of Records to read from Db in one go
payloadSize -> max size of payload that can be sent in one go to ES, max number of entities that can be sent = batchSize , or else whenever we hit the payload size it's flushed